src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -352,13 +363,19 @@ def rbd_pool_list(cls, pool_name, namespace=None):
                 namespaces.append('')
             for current_namespace in namespaces:
                 ioctx.set_namespace(current_namespace)
-                names = cls._rbd_image_names(ioctx)
-                for name in names:
+                image_refs = cls._rbd_image_refs(ioctx)
+                for image_ref in image_refs:
                     try:
-                        stat = cls._rbd_image_stat(ioctx, pool_name, current_namespace, name)
+                        stat = cls._rbd_image_stat(
+                            ioctx, pool_name, current_namespace, image_ref['name'])
                     except rbd.ImageNotFound:
-                        # may have been removed in the meanwhile
-                        continue
+                        # Check if the RBD has been deleted partially. This happens for example if
+                        # the deletion process of the RBD has been started and was interrupted.
+                        try:
+                            stat = cls._rbd_image_stat_removing(
+                                ioctx, pool_name, current_namespace, image_ref['id'])
+                        except rbd.ImageNotFound:
+                            continue
                     result.append(stat)
```

Comment:
A question triggered by seeing multiple `source == REMOVING` checks elsewhere: why are we appending to `result` without checking the source?

Assuming that everything appended to `result` gets displayed in the "Images" table, what if the user does `rbd trash mv <image name>` right after `_rbd_image_refs()` is called?  The source would be `USER` and it seems like the image that should be displayed in the "Trash" table (because it was trashed by the user) would be displayed in the "Images" table, potentially screwing up the UI because none of the expected image attributes would be there.

Reply:
> A question triggered by seeing multiple `source == REMOVING` checks elsewhere: why are we appending to `result` without checking the source?

Good question! - The "simple" answer is: I didn't know it better - I'm sorry :disappointed: 

> 
> Assuming that everything appended to `result` gets displayed in the "Images" table, what if the user does `rbd trash mv <image name>` right after `_rbd_image_refs()` is called? The source would be `USER` and it seems like the image that should be displayed in the "Trash" table (because it was trashed by the user) would be displayed in the "Images" table, potentially screwing up the UI because none of the expected image attributes would be there.

So I guess I should filter for "USER" RBDs, so they won't be returned? 

As a side-note: I don't see a difference in the returned data between RBDs with source == 'USER' and source == 'REMOVING'. I already took care of source == 'REMOVING' RBDs not breaking the UI because of missing attributes.

Reply:
The data is the same, but `source == USER` means that it was the user who moved the image to the trash.  It's not being removed (it may stay in the trash for days or weeks), so it shouldn't be listed in "Images" table.

Reply:
> The data is the same, but `source == USER` means that it was the user who moved the image to the trash. It's not being removed (it may stay in the trash for days or weeks), so it shouldn't be listed in "Images" table.

Thanks, understood, I'll update the PR. I just wanted to point out (for other reviewers) the UI is not broken if RBDs with source == 'REMOVING' are shown as part of the RBD list.